packages/karma-fixture/src/init.js

```diff
@@ -396,5 +396,20 @@ beforeEach(async () => {
 });
 
 afterEach(() => {
+  // Try to clean up any lingering media elements.
```

Comment:
I am still seeing warning in the log. 

```
[1208/203114.673605:INFO:CONSOLE(0)] "Blocked attempt to create a WebMediaPlayer as there are too many WebMediaPlayers already in existence. See crbug.com/1144736#c27", source: http://localhost:9876/?id=60207859#page=%25222a164aef-3fc7-40d2-a32e-dc916896de2c%2522 (0)
```

Reply:
Yes I know 😞

Reply:
Would it be possible to try cleaning it up generally with `onended` event listener, not just in `afterEach()`?
Not sure if it's possible to add general listeners like this.

Reply:
Wouldn't that break tests? because the media element is potentially removed in the middle of a test.

That would probably mean adding a mutation observer to detect any time a new media element is added and then attach an `ended` listener for that media element. And immediately removing an element at that po

Initial research indicates that cleanup like in this PR in an `afterEach` _should_ work.

Reply:
Hm, good point that this would potentially remove it in the middle of the test. Was thinking of alternatives since it doesn't seem to be working in `afterEach` for some reason 🤷

Reply:
The weird thing is that these warnings pop up already after the first few tests 🤔

Reply:
Maybe there's something off with how media is set up for each test?

Even if we'd be creating a lot of videos for each test, still seems very odd that the limit 10000 is exceeded after the first few tests 😕 

I wonder, when doing `document.querySelectorAll('video,audio')` inside a test and logging out the count, does it show unexpectedly high numbers?

Reply:
The amount of media items in the library is one thing, the number of actual video elements created because of re-renders is another. Chrome will create a new _player_ every time in that case.

You can see that by opening `chrome://media-internals/` in Chromium when running tests in non-headless and observing the log. There are tons and tons of players being created.

Most are discarded really quickly as tests progress, but some players seem to be lingering for up to a minute before being destroyed.

Reply:
cc @choumx perhaps you have some ideas here to help declutter the logs from these messages?

Reply:
Hmm #10023 might fix this actually

Reply:
oh nice. since it'll just render a poster for the video instead of the video?

yea I've been looking into the media players error in the karma logs. noticed when we start getting the `WebMediaPlayer` error it increments by 12 every time we remount the fixture cus theres 12 media items that are videos rendered in the media tab.

I couldn't figure out why the detached video nodes weren't getting GCed, but seems to be an issue for a lot of ppl with react/chrome:
https://github.com/facebook/react/issues/15583

Would be nice to figure out how to have less of a memory leak around dom nodes in general when restoring and remounting the fixture between tests. Definitely gets north of 100,000 nodes by some of the later tests if you run the performance monitor while the karma tests are running (and that's after forcing a garbage collection run in the memory tab)

I tried using [cleanup()](https://testing-library.com/docs/react-testing-library/api/#cleanup) in our `restore` method for the fixture, but still wasn't quite doing the trick

Reply:
This is right after forcing GC and having the karma tests run for 20 minutes or so
<img width="1439" alt="Screen Shot 2022-01-06 at 1 19 46 PM" src="https://user-images.githubusercontent.com/35983235/148438965-7fec8c6a-6c66-434b-ab48-58b43e713fbc.png">

Reply:
I'm looking at a lot of the leftover `Detached HTMLDivElement` that should be cleaned up after the fixture is restored. `SelectionCanvas` and `updateLasso` seem to come up a lot when inspecting references. Seems to be the only thing for a lot of them beyond react internals.

I'm curious if there's an event listener that's not getting cleaned up or some type of closure issue that's leaking and holding onto references here.

Seems to be an area of the codebase with a high probability for this to happen cus there's a lot going on with regard to event listeners, closures and refs

Reply:
Maybe it's not from the `SelectionCanvas` specifically 🤷 . Commented all the event handlers out and a lot of detached HTMLDivElements are still being kept around from other stuff.